txpool/txpool.go

```diff
@@ -659,13 +659,13 @@ func (p *TxPool) validateTx(tx *types.Transaction) error {
 
 			return ErrUnderpriced
 		}
-	} else {
-		// Legacy approach to check if the given tx is not underpriced
-		if tx.GetGasPrice(baseFee).Cmp(big.NewInt(0).SetUint64(p.priceLimit)) < 0 {
-			metrics.IncrCounter([]string{txPoolMetrics, "underpriced_tx"}, 1)
+	}
 
-			return ErrUnderpriced
-		}
+	// Check if the given tx is not underpriced
```

Comment:
Just to check, we are using baseFee for both legacy and dynamic transactions when we want to check if is it underpriced.

Reply:
Not really. Base fee value is needed for dynamic fee tx. Take a look at the function implementation to understand how the gas price is calculated for different tx types.

Reply:
`GetGasPrice` on the top has this:
```
if t.GasPrice != nil && t.GasPrice.BitLen() > 0 {
		return new(big.Int).Set(t.GasPrice)
```
so, for legacy `baseFee` parameter does not change anything

Reply:
Now I do not understand where is the part of the code which checks if a legacy transaction is underpriced?

Reply:
`GetGasPrice` has the below condition, when can baseFee be 0?

```
} else if baseFee == 0 {
		return new(big.Int)
	}
```
and Please update the comment on top of `GetGasPrice` method from `- or calculate a value with formula: min(gasFeeCap, gasTipCap * baseFee);` to `- or calculate a value with formula: min(gasFeeCap, gasTipCap + baseFee);`